astropy/table/column.py

```diff
@@ -746,6 +746,17 @@ def __array_wrap__(self, out_arr, context=None, return_scalar=False):
             out_arr = super().__array_wrap__(out_arr, context)
         else:
             out_arr = super().__array_wrap__(out_arr, context, return_scalar)
+
+            # force return type to pure numpy arrays for certain ufuncs
+            # this is consistent with numpy 1.x behaviour for these functions
+            _function = context[0] if context is not None else None
+            if _function in (np.isfinite, np.isinf, np.isnan, np.sign, np.signbit):
+                if isinstance(out_arr, MaskedColumn):
+                    view_type = np.ma.MaskedArray
+                else:
+                    view_type = np.ndarray
+                out_arr = out_arr.view(view_type)
+
         if (NUMPY_LT_2_0 or return_scalar) and (
```

Comment:
Actually, the problem is that the previous fix caused a breakage: `np.isfinite` is in `_comparison_functions`. So, we need to combine the two, as follows:
```
if self.shape != out_arr.shape or (
        isinstance(out_arr, BaseColumn)
        and (context is not None and context[0] in _comparison_functions)
    ):
    data = out_arr.data
    return data[()] if NUMPY_LT_2_0 or return_scalar else data
else:
    return out_arr
```

Reply:
I'm not sure what breakage you've identified, and see no failing test with this branch (after rebase) locally.
For completion, I *also* don't see any failures with your suggestion applied.
I'll let CI complete here to see if I'm missing anything.

Reply:
Ok, all I see is ✅, and it's night time in France, so I'm logging off, but feel free to push your suggestion to my branch if needed ! Thank you !